Sources/TestsCommon/CommonTestUtils.swift

```diff
@@ -208,3 +208,22 @@ extension TopologyDescription.TopologyType {
         }
     }
 }
+
+public struct TestError: LocalizedError {
+    public var message: String
+    public var errorDescription: String { return self.message }
+
+    public init(message: String) {
+        self.message = message
+    }
+}
+
+/// Possible authentication mechanisms.
+public enum AuthMechanism: String, Decodable {
```

Comment:
this is used by both the sync and async auth tests so I put it here.
eventually this will be moved out into the driver itself as part of the public API.